lib/travis/build/addons/browserstack.rb

```diff
@@ -0,0 +1,153 @@
+              sh.echo "You provided an invalid access_key.", ansi: :red
+              return
+            end
+
+            sh.echo "Installing BrowserStack Local", ansi: :yellow
+            sh.if "$(uname) = 'Linux'" do
+              bin_package = "#{BROWSERSTACK_BIN_FILE}-linux-x64.zip"
+              bin_url = "#{BROWSERSTACK_BIN_URL}/#{bin_package}"
+
+              sh.cmd "mkdir -p #{BROWSERSTACK_HOME}"
+              sh.cmd "wget -O /tmp/#{bin_package} #{bin_url}", echo: true, timing: true, retry: true
+              sh.cmd "unzip -d #{BROWSERSTACK_HOME}/ /tmp/#{bin_package} 2>&1 > /dev/null", echo: false
+              sh.chmod "+x", "#{BROWSERSTACK_HOME}/#{BROWSERSTACK_BIN_FILE}", echo: false
+            end
+            sh.else do
+              sh.echo "Unsupported platform: $TRAVIS_OS_NAME.", ansi: :yellow
+            end
+          end
+
+          sh.fold 'browserstack.local.start' do
+            sh.echo 'Starting BrowserStack Local', ansi: :yellow
+            sh.cmd "#{build_start_command(browserstack_key)}"
+            sh.export ENV_KEY, browserstack_key, echo: false
+            sh.export ENV_LOCAL, 'true', echo: false
+          end
+        end
+
+        def after_after_script
+          sh.if "-f #{BROWSERSTACK_HOME}/#{BROWSERSTACK_BIN_FILE}" do
+            sh.fold 'browserstack.local.stop' do
+              sh.echo 'Stopping BrowserStack Local', ansi: :yellow
+              sh.cmd "#{build_stop_command}"
+            end
+          end
+        end
+
+        def build_start_command(access_key)
+          args = []
+          options = {
+            :localIdentifier => method(:local_identifier),
+            :v => method(:verbose),
+            :f => method(:folder),
+            :force => method(:force),
+            :only => method(:only),
+            :forceLocal => method(:force_local),
+            :onlyAutomate => method(:only_automate),
+            :proxyHost => method(:proxy_host),
+            :proxyPort => method(:proxy_port),
+            :proxyUser => method(:proxy_user),
+            :proxyPass => method(:proxy_pass)
+          }
+
+          options.each do |arg, fn|
+            v = fn.call
+            if !v.nil?
+              if (!!v == v)
+                args.push("-#{arg}") if v
+              else
+                args.push("-#{arg}")
+                args.push(v)
+              end
+            end
+          end
+
+          local_args = args.empty? ? "" : " #{args.join(' ')}"
+          "#{BROWSERSTACK_HOME}/#{BROWSERSTACK_BIN_FILE} -d start #{access_key}#{local_args}" if !access_key.nil?
+        end
+
+        def build_stop_command
+          "#{BROWSERSTACK_HOME}/#{BROWSERSTACK_BIN_FILE} -d stop"
+        end
+
+        private
+          def access_key
+            key = config[:access_key] || config[:accessKey]
+            (!key.nil? && key.match(/^[a-zA-Z0-9]+$/)) ? key : nil
+          end
+
+          def verbose
+            v = config[:verbose] || config[:v]
```

Comment:
Please avoid using shorthand configuration parameters such as `v` and `f` (below).